x/tally/keeper/endblock.go

```diff
@@ -103,20 +103,26 @@ func (k Keeper) ProcessTallies(ctx sdk.Context, coreContract sdk.AccAddress) err
 			return fmt.Errorf("invalid gas price: %s", req.GasPrice)
 		}
 
-		var record types.PayoutRecord
+		var gasCalc GasCalculation
 		switch {
 		case len(req.Commits) < int(req.ReplicationFactor):
 			dataResults[i].Result = []byte(fmt.Sprintf("need %d commits; received %d", req.ReplicationFactor, len(req.Commits)))
 			dataResults[i].ExitCode = TallyExitCodeNotEnoughCommits
 			k.Logger(ctx).Info("data request's number of commits did not meet replication factor", "request_id", req.ID)
-			record.ExecDists = k.CalculateCommitterPayouts(ctx, req, params.GasCostCommit, gasPrice)
+
+			gasCalc.Executors, dataResults[i].GasUsed = CommitGasUsed(req, params.GasCostCommit, req.ExecGasLimit)
+			dataResults[i].GasUsed += params.GasCostBase
+			gasCalc.Burn += params.GasCostBase
 		case len(req.Reveals) == 0:
 			dataResults[i].Result = []byte("no reveals")
 			dataResults[i].ExitCode = TallyExitCodeNoReveals
 			k.Logger(ctx).Info("data request has no reveals", "request_id", req.ID)
-			record.ExecDists = k.CalculateCommitterPayouts(ctx, req, params.GasCostCommit, gasPrice)
+
+			gasCalc.Executors, dataResults[i].GasUsed = CommitGasUsed(req, params.GasCostCommit, req.ExecGasLimit)
+			dataResults[i].GasUsed += params.GasCostBase
+			gasCalc.Burn += params.GasCostBase
```

Comment:
I'm not sure if this case makes sense? If there's no reveals, but we know that there are enough commits to meet the replication factor then there's either a replication factor of 0 (which the contract doesn't allow if I'm not mistaken), or all executors are refusing to reveal which should result in slash events for all of them.

Reply:
We should probably remove this case. However that won't change the executor payout because we pay commit gas to the committers when basic consensus is not reached.

Reply:
Perhaps we should change the logic of `CommitGasUsed` so that we only pay the commit gas to the executors that have both committed AND revealed? If someone has committed but failed to reveal, we don't pay the person anything.

Reply:
Yes, except for the case where there are no reveals at all (and the replication factor wasn't met). In that scenario we still attribute them the fallback gas consumption.

Reply:
Addressed this issue in https://github.com/sedaprotocol/seda-chain/pull/464/commits/10fd40b64d57075372ee6fdcba435fef165f046e Please let me know if you still see some problems